boa/src/builtins/object/mod.rs

```diff
@@ -410,6 +410,18 @@ pub fn make_object(_: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultVa
     Ok(object)
 }
 
+/// Creates a new object from the provided prototype
+///
+/// https://tc39.es/ecma262/#sec-object.create
+/// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create
+pub fn create_builtin(_: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+    let __proto__ = args.get(0).cloned().unwrap_or_else(Value::undefined);
```

Comment:
I think we need to make sure `__proto__` has the type `object` or `null` before moving on.